components/vocabularies/oncotree/api/src/main/java/org/phenotips/vocabulary/internal/solr/OncoTree.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.vocabulary.internal.solr;
+
+import org.phenotips.vocabulary.VocabularyTerm;
+
+import org.xwiki.component.annotation.Component;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.DisMaxParams;
+import org.apache.solr.common.params.SpellingParams;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Provides access to the OncoTree vocabulary. The vocabulary prefix is {@code ONCO}.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component
+@Named("onco")
+@Singleton
+public class OncoTree extends AbstractCSVSolrVocabulary
+{
+    /** The default location of the OncoTree data file. */
+    private static final String SOURCE = "http://oncotree.mskcc.org/oncotree/api/tumor_types.txt";
```

Comment:
This is probably too much work for this issue, but may be useful to create an issue to make this configurable in the admin section or in a config file if this is only used once during first startup